### PR TITLE
Regex is better at validating URLs; can also post gifs :)

### PR DESCRIPTION
**Pull request by jhuang09**

### CODE REVIEW

**Review by dhgarrette — CHANGES_REQUESTED** [review]
Nice!  Since this regex is more complicated than the previous one, can you add a comment directly above it that explains what kinds of patterns it captures?

**Comment by jhuang09** [comment]
I've added a comment explaining its pattern. It seems a bit complicated and I'm not sure if it can be made clearer. Thoughts?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 1]
```diff
@@ -47,8 +47,15 @@ public void init() {
    * properly.
    */
   public void prepareMessageForDisplay(Message message) {
-    String regex = "(https?://\\S+\\.(png|jpg))";
-    String replacement = "<img src=\"$1\" />";
+    /**
```
Comment (dhgarrette) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: I think it's pretty good, but I have two comments:
1) "Javadoc" style comments `/**  */` are for documenting methods and classes, but regular comments should use `//`.
2) I think you might be able to simplify slightly by giving a real example, and a little bit of prose description.  Maybe something like:

```
// Matches the URL of an image file, with an optional caption. For example:
//     [the google logo] http://www.google.com/images/logo.png
// Matched URLs must end with one of: .png, .jpg, .gif
String regex = "(\\[.+\\])?\\s(https?://[\\S+\\.]+/([\\S+\\.?]+/?)+\\.(png|jpg|gif))";

// Replaces the URL with the actual image. If a caption was given, it is printed below the image.
String replacement = "<figure><img src=\"$2\" /> <figcaption>$1</figcaption></figure>";
```

**Review by dhgarrette — APPROVED** [review]
Thanks, Janny!

**Comment by jhuang09** [comment]
I've updated the comments, but I realized that there's some faulty logic with my regex that I will fix soon! Will update once I get a better regex

**Comment by dhgarrette** [comment]
I merged my PR with the unit test, so you can update that as needed to add test cases for your change.

**Pull request merged**